sim/hunter/runes.go

```diff
@@ -364,18 +371,18 @@ func (hunter *Hunter) trapRange() float64 {
 	return 5
 }
 
-func (hunter *Hunter) resourcefulnessManacostModifier() float64 {
-	if hunter.HasRune(proto.HunterRune_RuneCloakResourcefulness) {
-		return 0.0
+func (hunter *Hunter) applyResourcefulness() {
+	if !hunter.HasRune(proto.HunterRune_RuneCloakResourcefulness) {
+		return
 	}
-	return 1.0
-}
 
-func (hunter *Hunter) resourcefulnessCooldownModifier() float64 {
-	if hunter.HasRune(proto.HunterRune_RuneCloakResourcefulness) {
-		return 0.6
-	}
-	return 1.0
+	hunter.OnSpellRegistered(func(spell *core.Spell) {
+		if spell.Flags.Matches(SpellFlagTrap) {
+			// Ideally I would set Multiplier to 0 but that results in the cost calculation defaulting to 100
+			spell.Cost.FlatModifier = int32(spell.Cost.BaseCost * -1)
+			spell.CD.Duration = spell.CD.Duration / 100 * 60
+		}
```

Comment:
Couldn't you just change `spell.Cost.BaseCost` to 0?

Reply:
Oh, I guess idk I thought base cost was different than  flat cost but now I'm wondering if I'm confusing 2 different structs

Reply:
They are. Flat cost is a raw mana cost (e.g. 300) and Base cost is a % of total mana. It looks like traps are a flat cost so actually youd want to do `spell.Cost.FlatCost = 0` if you can

Reply:
I think that's where I was confused, I'm not sure FlatCost exists outside of the spell configuration SpellConfig.ManaCost struct. Seems like setting the Cost.BaseCost of the spell post construction is the correct way to set it.

Reply:
Oh I see. Ok that's fine then